### PR TITLE
[4.2] [stdlib] Revise documentation

### PR DESCRIPTION
**Pull request by natecook1000**
This includes various revisions to recently added standard library APIs, including:
- Random and other randomness APIs
- Hashable changes
- MemoryLayout.offset(of:)

Cherry picked from #16707.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please test

**Comment by natecook1000** [comment]
Note: the one non-documentation change in this is switching from `@_inlineable` to `@inlinable` for `MemoryLayout.offset(of:)`.

File: docs/Random.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+The implementation of the default random generator varies by platform. The implementation
+on each platform must be thread-safe and automatically seeded, and should be
+cryptographically secure to the extent possible. Currently supported platforms have the
+following implementation details:
+
+- Apple platforms use `arc4random_buf(3)`.
+- Linux, FreeBSD, and other UNIX-like platforms use `getrandom(2)` when available;
+otherwise, they read from `/dev/urandom`.
+- Fuchsia platforms use `getentropy(3)`.
+- Windows paltforms use `BCryptGenRandom`.
```
Comment (Azoy) [inline docs/Random.md]: platforms*

Comment (Azoy) [inline docs/Random.md]: platforms*

Comment (natecook1000) [reply docs/Random.md]: Thanks!

**Comment by natecook1000** [comment]
@swift-ci Please test and merge

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**